ultra/docs/curriculum_learning.md

```diff
@@ -0,0 +1,126 @@
+
+Unlike the regular the training setup (curriculum mode disable), there is another task folder called curriculum under ultra/scenarios which has the blueprint of which set of scenarios are in each grade
+
+```yaml
+curriculum:
+  static:
+    toggle: <bool> # Turn on static curriculum settings
+    grades:
+      1: [[<task>,<level>]]
+      2: [[<task>,<level>], [<task>,<level>]]
+      # 3: [[<task>,<level>], ... ,[<task>,<level>]] can have as many combinations of tasks and levels
+    conditions:
+      eval_per_grade: <bool> # Evaluation takes places after completion of grade. An "exam" at the end of year
+      episode_based: # Agent graduates after completing a N number of episodes in each grade
+        toggle: <bool> # Enable the condition
+        cycle: <bool> # Option to keep cycling through grades to episodes limit
+      pass_based: # Agent graduates after getting an average completion rate, the average is taken over the eval-rate (sampling-rate)
+        toggle: <bool> # Enable the condition
+        pass_rate: <float> # Scalar between 0 and 1; describes the threshold completion rate (%)
+        sample_rate: <int> # Samples for some parameter (average_reached_goal) at every N episodes
+        warmup_episodes: <int> # Starting N episodes where sampling does not occur, thus no graduation can take place
+```
+A more specific example in which we take the three levels from task 1 and distribute it among three grades
+
+```yaml
+curriculum:
+  static:
+    toggle: True
+    grades:
+      1: [[1,no-traffic]]
+      2: [[1,easy]]
+      3: [[1,hard]]
+      conditions:
+        eval_per_grade: True
+        episode_based:
+          toggle: True 
+          cycle: True 
+        pass_based:
+          toggle: False 
+          pass_rate: 0.50 # If the average scenario passed exceeds more than 0.5 then grade is switched
+          sample_rate: 30 # Every 30 episodes slot, the average reached goal (arg) is calculated (arg = total_scenario_passed / sample_rate)
+          warmup_episodes: 100 # Starting at any grade, the first 100 episodes will be subjected to no sampling
+```
+Now when we enter grade 1, the agent will only face the scenarios that are part of (task 1, level easy), same applies to grades 2 and 3. The conditions for graduation is that the agent will complete N number of episodes in each grade. N is the quotient of the total number of episodes / total number of grades. The other condition is that agents will only graudate from a grade if it has reached a specified "passing mark". The "passing mark" can be any param (env_score, reached_goal, collisions) that is aggregated over some N sample of episodes (which is defined in the curriculum config file).
```

Comment:
```suggestion
Now when we enter grade 1, the agent will only face the scenarios that are part of (task 1, level no-traffic). Grade 2 consists of scenarios that art part of (task 1, level easy), and grade 3 has scenarios from (task 1, level hard). The conditions for graduation to the next grade is that the agent will complete N number of episodes in each grade. N is the quotient of the total number of episodes / total number of grades. The other condition is that agents will only graduate from a grade if it has reached a specified "passing mark". The "passing mark" can be any metric (`env_score`, `reached_goal`, `collisions`) that is averaged over N sample of episodes (which is defined in the curriculum config file).
```
Is "(task 1, level easy)" supposed to be "(task 1, level no-traffic)"?
Is "aggregated" supposed to be "averaged"?